partition_snapshot_row_cursor.hh

```diff
@@ -752,66 +753,71 @@ public:
         return _position;
     }
 
-    friend std::ostream& operator<<(std::ostream& out, const partition_snapshot_row_cursor& cur) {
-        fmt::print(out, "{{cursor: position={}, cont={}, rt={}}}",
+    friend fmt::formatter<partition_snapshot_row_cursor>;
+};
+
+inline
+partition_snapshot_row_weakref::partition_snapshot_row_weakref(const partition_snapshot_row_cursor& c)
+    : _it(c._current_row[0].it)
+    , _change_mark(c._change_mark)
+    , _pos(c._position)
+    , _in_latest(c.is_in_latest_version())
+{ }
+
+inline
+partition_snapshot_row_weakref& partition_snapshot_row_weakref::operator=(const partition_snapshot_row_cursor& c) {
+    auto tmp = partition_snapshot_row_weakref(c);
+    this->~partition_snapshot_row_weakref();
+    new (this) partition_snapshot_row_weakref(std::move(tmp));
+    return *this;
+}
```

Comment:
why was this code moved?

Reply:
it *looks like* this piece of code was moved. but what happened was, in order to define the full specialization of `fmt::formatter` for `partition_snapshot_row_cursor`, i had to define it out of this class. so as usual i put the template at the end of the file. so it seems i was moving this piece of code around.

anyway, will define the template where we defined the operator<< in the next revision, so we have less code churn.